ypy_websocket/django_channels/yroom_storage.py

```diff
@@ -0,0 +1,226 @@
+import time
+from typing import Optional
+
+import redis.asyncio as redis
+import y_py as Y
+
+
+class BaseYRoomStorage:
+    """Base class for YRoom storage.
+
+    This class is responsible for storing, retrieving, updating and persisting the Ypy document.
+
+    Each Django Channels Consumer should have its own YRoomStorage instance, so each consumer is connected to the same document.
+
+    Updates to the document should be sent to the shared storage, instead of each consumer having its own version of the YDoc.
+
+    A full example of a Redis as temporary storage and Postgres as persistent storage is:
+
+    ```py
+    from typing import Optional
+
+    from django.db import models
+    from ypy_websocket.django_channels.yroom_storage import RedisYRoomStorage
+
+
+    class YDocSnapshotManager(models.Manager):
+        async def aget_snapshot(self, name) -> Optional[bytes]:
+            try:
+                instance: YDocSnapshot = await self.aget(name=name)
+                result = instance.data
+                if not isinstance(result, bytes):
+                    # Postgres on psycopg2 returns memoryview
+                    return bytes(result)
+
+            except YDocSnapshot.DoesNotExist:
+                return None
+            else:
+                return result
+
+        async def asave_snapshot(self, name, data):
+            return await self.aupdate_or_create(name=name, defaults={"data": data})
+
+
+    class YDocSnapshot(models.Model):
+        name = models.CharField(max_length=255, primary_key=True)
+        timestamp = models.DateTimeField(auto_now=True)
+        data = models.BinaryField()
+
+        objects = YDocSnapshotManager()
+
+        def __str__(self):
+            return self.name
+
+
+    class CustomRoomStorage(RedisYRoomStorage):
+        async def get_snapshot_from_database(self) -> Optional[bytes]:
+            return await YDocSnapshot.objects.aget_snapshot(self.room_name)
+
+        async def save_snapshot(self):
+            current_snapshot = await self.redis.get(self.redis_key)
+
+            if not current_snapshot:
+                return
+
+            await YDocSnapshot.objects.asave_snapshot(
+                self.room_name,
+                current_snapshot,
+            )
+
+    ```
+    """
+
+
+    def __init__(self, room_name) -> None:
+        self.room_name = room_name
+
+        self.last_saved_at = time.time()
+        self.save_throttled_interval = 5
```

Comment:
I would usually call this
```suggestion
        self.save_throttle_interval = 5
```
WDYT?

Reply:
agree!